experimental/python2.7/testdata/python27.yaml

```diff
@@ -23,588 +23,48 @@ commandTests:
-    exitCode: 0
-  - name: import_uu
-    command: ["/usr/bin/python2.7", "-c", "import uu"]
-    exitCode: 0
-  - name: import_uuid
-    command: ["/usr/bin/python2.7", "-c", "import uuid"]
-    exitCode: 0
-  - name: import_warnings
-    command: ["/usr/bin/python2.7", "-c", "import warnings"]
-    exitCode: 0
-  - name: import_wave
-    command: ["/usr/bin/python2.7", "-c", "import wave"]
-    exitCode: 0
-  - name: import_weakref
-    command: ["/usr/bin/python2.7", "-c", "import weakref"]
-    exitCode: 0
-  - name: import_webbrowser
-    command: ["/usr/bin/python2.7", "-c", "import webbrowser"]
-    exitCode: 0
-  - name: import_whichdb
-    command: ["/usr/bin/python2.7", "-c", "import whichdb"]
-    exitCode: 0
-  - name: import_wsgiref
-    command: ["/usr/bin/python2.7", "-c", "import wsgiref"]
-    exitCode: 0
-  - name: import_xdrlib
-    command: ["/usr/bin/python2.7", "-c", "import xdrlib"]
-    exitCode: 0
-  - name: import_xml
-    command: ["/usr/bin/python2.7", "-c", "import xml"]
-    exitCode: 0
-  - name: import_xmlrpclib
-    command: ["/usr/bin/python2.7", "-c", "import xmlrpclib"]
-    exitCode: 0
-  - name: import_zipfile
-    command: ["/usr/bin/python2.7", "-c", "import zipfile"]
-    exitCode: 0
-  - name: import_zipimport
-    command: ["/usr/bin/python2.7", "-c", "import zipimport"]
-    exitCode: 0
+  # import every module installed with the Python package
+  - name: import_everything
+    exitCode: 0
+    command:
+      - "/usr/bin/python2.7"
+      - "-c"
+      # multi-line YAML string with Python script that imports all modules that are installed.
+      # This ensures we have the right native library dependencies.
+      - |
+        import pkgutil
+
+        skip_modules = frozenset((
+          # Windows-specific modules
+          'ctypes.wintypes',
+          'distutils.command.bdist_msi',
+          'distutils.msvc9compiler',
+          'encodings.mbcs',
+
+          # depends on things not installed by default on Debian
+          'lib2to3.pgen2.conv',
+          '_LWPCookieJar',
+          '_MozillaCookieJar',
+        ))
+
+        def onerror(failed_module_name):
+          # ignore any errors from modules we are explicitly skipping (e.g. tkinter)
+          if failed_module_name in skip_modules:
+            return
+          raise Exception('failed to import module: {}'.format(repr(failed_module_name)))
+
+        for module_info in pkgutil.walk_packages(onerror=onerror):
+          # TODO: use module_info.name when Debian9 support is dropped
+          module_name = module_info[1]
+          # idlelib on Python3.5 depends on tkinter which is not installed by default
+          if module_name in skip_modules or module_name.endswith('.__main__'):
+            continue
+
+          try:
+            __import__(module_name)
+          except ImportError as e:
+            # ignore tkinter import errors: there are so many
+            if 'No module named _tkinter' in str(e):
```

Comment:
Can this not be in `skip_modules` or handled in `onerror()`?

Reply:
See my previous comment: walk_packages does not actually import everything. It only imports "packages", so we need the error handler in two places effectively.